src/mlpack/core/optimizers/adam/amsgrad_update.hpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file amsgrad_update.hpp
+ * @author Haritha Nair
+ *
+ * Implementation of AMSGrad optimizer. AMSGrad is an exponential moving average 
+ * optimizer that dynamically adapts over time with guaranteed convergence.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_AMS_GRAD_AMS_GRAD_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_AMS_GRAD_AMS_GRAD_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * AMSGrad is an exponential moving average variant which along with having 
+ * benefits of optimizers like Adam and RmsProp, also guarantees convergence.
+ * Unlike adam, it uses maximum of past squared gradients rather than their 
+ * exponential average for updation.
```

Comment:
This is a really pedantic comment just for consistency---can you capitalize the same way the mlpack classes are?  So `adam` -> `Adam`, `RmsProp` -> `RMSProp`.